Libplanet/Blockchain/BlockChain.cs

```diff
@@ -250,96 +250,81 @@ IEnumerator IEnumerable.GetEnumerator()
             }
 
             Block<T> block = Blocks[offset.Value];
+            Tuple<HashDigest<SHA256>, long> stateReference;
 
-            ImmutableHashSet<Address> requestedAddresses =
-                addresses.ToImmutableHashSet();
-            var stateReferences = new HashSet<Tuple<HashDigest<SHA256>, long>>();
-
-            foreach (var address in requestedAddresses)
+            _rwlock.EnterReadLock();
+            try
             {
-                Tuple<HashDigest<SHA256>, long> sr;
-                _rwlock.EnterReadLock();
-                try
-                {
-                    sr = Store.LookupStateReference(Id, address, block);
-                }
-                finally
-                {
-                    _rwlock.ExitReadLock();
-                }
+                stateReference = Store.LookupStateReference(Id, address, block);
+            }
+            finally
+            {
+                _rwlock.ExitReadLock();
+            }
 
-                if (!(sr is null))
-                {
-                    stateReferences.Add(sr);
-                }
+            if (stateReference == null)
```

Comment:
We're using pattern matching for checking null.

```suggestion
            if (stateReference is null)
```

Reply:
Nice find. Thank you.